### PR TITLE
HPCC-14428 Replace popen/pclose with _popen/_pclose on Windows

### PR DESCRIPTION
**Pull request by xwang2713**
HPCC 6.0.0 Platform builds failed on Windows due to popen/pclose not defined and dllimport is not allowed in *.cpp file

Provided fixes resolve these issues by replacing popen/pclose with _poen/_pclose on Windows and 
removing "extern COMPONENTSTATUS_API" in componentstatus.cpp

Signed-off-by: Xiaoming Wang xiaoming.wang@lexisnexi.com 

@wangkx please reivew

### CODE REVIEW

**Comment by wangkx** [comment]
@xwang2713 looks fine.

**Comment by xwang2713** [comment]
@richardkchapman this pull-request is ready to merge. The problem originally posted from HPCC forum: https://hpccsystems.com/bb/viewtopic.php?f=14&t=1950&p=8372#p8372

**Comment by richardkchapman** [comment]
There's a typo in the commit message title.

Normally we try to avoid putting os-specific code outside jlib - e.g. using a macro in platform.h might be a better approach.

File: esp/services/ws_machine/componentstatus.cpp [hunk 1]
```diff
@@ -331,7 +331,7 @@ static CComponentStatusFactory *csFactory = NULL;
 
 static CriticalSection getComponentStatusSect;
 
-extern COMPONENTSTATUS_API IComponentStatusFactory* getComponentStatusFactory()
```
Comment (richardkchapman) [inline esp/services/ws_machine/componentstatus.cpp]: Not clear why this is the right fix. Why is this function declared extern "C" in the header file?

**Comment by richardkchapman** [comment]
@xwang2713 Back to you

**Comment by xwang2713** [comment]
Move os-specific code to platform.h
Remove 'Extern "C" ' in componentstatus.hpp

**Comment by richardkchapman** [comment]
@wangkx There seems to be an unrelated commit included. Could do with rebasing/squashing too...

**Comment by xwang2713** [comment]
Rebased: removed unrelated changes and commit message.

**Pull request merged**